### ISSUE TITLE
Spike for reducing coupling between tracker featurizer, state featurizer, domain and policy

### ISSUE THREAD

**Issue by TyDunn** [RasaHQ/rasa#9049]
We discussed reducing coupling between tracker featurizer, state featurizer, domain and policy in Rasa Open Source 3.0. However, we don't have enough information to prioritise this with confidence and include it as part of the 3.0 action plan. We are aiming to make a decision on if we move forward with this based on the outcome of this spike by **Thu, Jul 15th**.

**Definition of done**
- [x] Problems with current coupling clearly defined
- [x] State of the codebase at the end of this decoupling articulated
- [x] How will we know that we've achieved it described
- [x] What would be needed of research and engineering explained
- [x] Risks of decoupling identified (in parallel with architecture revamp)
- [x] Enable and research team members have reviewed this document

**Probably out of scope of this spike but might be needed to decide re: 3.0?**
- [ ] Prototype (quick and dirty approach)
- [ ] Implementation proposal

**Comment by TyDunn** [comment]
document [here](https://www.notion.so/rasa/Disentangling-Trackers-Domain-Policies-and-Featurizers-8bde1be25c7344029fd3fc21d7259f46)
